### PR TITLE
Add prettier

### PR DESCRIPTION
**Pull request by anandthakker**
Adds [prettier](https://prettier.io/) to auto-format source code, and use [lint-staged](https://www.npmjs.com/package/lint-staged) to automatically set up a git precommit hook that runs prettier on any changed files and runs `make lint` before each commit. 

Running `npm install` sets up the precommit hook.

### CODE REVIEW

File: .husky/pre-commit [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/bin/sh
+. "$(dirname "$0")/_/husky.sh"
+
+# run prettier on staged files
+npx lint-staged
```
Comment (jwmerrill) [inline .husky/pre-commit:5]: I thought `npx` was for running `npm ` binaries. I would have expected this to be `npm run lint-staged`. Is this an alternate usage of `npx`?

**Review by jwmerrill — COMMENTED** [review]
Can we also add either an npm or make task to check whether prettier would modify anything so that we can hook that into CI?

File: .husky/pre-commit (hunk 1 above)
Comment (anandthakker) [reply .husky/pre-commit:5]: Nope, it's running the `lint-staged` npm binary

**Comment by jwmerrill** [comment]
Can we have
```
"singleQuote": true,
```
in prettierrc? I think MathQuill had a pretty consistent convention of using single quotes before this, and we might as well stick to it.

**Comment by anandthakker** [comment]
Updated to single quotes

**Review by jwmerrill — APPROVED** [review]

**Pull request merged**